tests/ui/test_settings_enable_2fa.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+from playwright.sync_api import Page, expect
+from tenacity import RetryCallState, retry, stop_after_attempt, wait_random_exponential
+
+BASE_URL = "http://127.0.0.1:8080"
+
+
+def log_retry_error(retry_state: RetryCallState) -> None:
+    if retry_state is None or retry_state.outcome is None:
+        return
+    print(f"Retrying due to error: {retry_state.outcome.exception()}")
+
+
+@retry(
```

Comment:
Required retrying initial loading of `localhost:8080/login` because it seems the playwright's browser can spin up before the app is ready to serve requests.

Reply:
If we use pytest fixtures, we can have a liveness check on the app so that it doesn't even return/yield said fixture until it's usable.